### ISSUE TITLE
Flags break between two trigger_momentum_limitmovement triggers

### ISSUE THREAD

**Issue by McGravel** [momentum-mod/game#988]
**Describe the bug**
If you have two close-enough/touching `trigger_momentum_limitmovement` triggers and move between them, any flags (limited movement options) that they share will become unblocked despite the player still being within a `trigger_momentum_limitmovement` trigger entity with them set as limited.

**To Reproduce**
Have two close-enough/touching `trigger_momentum_limitmovement` triggers with at least one matching flag. 
The common flag(s) will become unblocked when moving to the other trigger.

Video of the bug, excuse the terrible typing. Just take note of the UI on the bottom right.
[Video](https://www.youtube.com/watch?v=lfJ-CZNwK9Q)

**Expected behavior**
Common flags should not be cleared when moving between triggers.

**Screenshots**

**Desktop/Branch (please complete the following information):**
 - OS: Windows
 - Branch: Latest develop branch

**Additional context**
Found randomly whilst testing some unrelated thing. Likely a low priority bug.

**Comment by Gocnak** [comment]
Good report! I see what's happening, we had a similar problem with slide triggers before too. Since you enter the trigger before leaving the old one, you're inside the other while the previous has its keys "unblocked".

Solution is to just keep applying the blocked keys OnThink in the limitmovement trigger.

**Comment by harshavb** [comment]
https://github.com/momentum-mod/game/pull/1009
